src/OracleService/Protocols/OracleHttpsProtocol.cs

```diff
@@ -51,6 +51,7 @@ public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, Cancellati
             HttpResponseMessage message;
             try
             {
+                int redirects = 2;
```

Comment:
Hi @shargon, could it be a merge between max redirects and timeout?
And sum the time taken between redirects?

Anyway, this will solve the issue for now.

Reply:
Then we should create a new `HttpClient` for each request, because the timeout it's specified in the class

Reply:
Why not use [httpclienthandler.maxautomaticredirections](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.maxautomaticredirections?view=net-6.0#system-net-http-httpclienthandler-maxautomaticredirections) ?

Reply:
Because the original issue, we can have a SSRF